numpy/core/src/multiarray/convert_datatype.c

```diff
@@ -679,15 +679,62 @@ NPY_NO_EXPORT npy_bool
 PyArray_CanCastTypeTo(PyArray_Descr *from, PyArray_Descr *to,
                                                     NPY_CASTING casting)
 {
-    /* Fast path for unsafe casts or basic types */
-    if (casting == NPY_UNSAFE_CASTING ||
-            (NPY_LIKELY(from->type_num < NPY_OBJECT) &&
+    /*
+     * Fast path for basic types.
+     */
+    if ((NPY_LIKELY(from->type_num < NPY_OBJECT) &&
```

Comment:
I suppose `<=` would work as well, not that it matters.

Reply:
This was there before, but you're right, no sense in not including object.